File: Android/app/src/main/java/com/secureauth/models/TOTPAccount.kt
package com.secureauth.models

import java.util.UUID

data class TOTPAccount(
    val id: String = UUID.randomUUID().toString(),
    val issuer: String,
    val accountName: String,
    val secret: String,
    val algorithm: String = "SHA1",
    val digits: Int = 6,
    val period: Int = 30
)

File: iOS/SecureAuth/Models/TOTPAccount.swift
import Foundation

struct TOTPAccount: Identifiable, Codable {
    let id: UUID
    let issuer: String
    let accountName: String
    let secret: String
    let algorithm: String
    let digits: Int
    let period: Int

    init(id: UUID = UUID(), issuer: String, accountName: String, secret: String, algorithm: String = "SHA1", digits: Int = 6, period: Int = 30) {
        self.id = id
        self.issuer = issuer
        self.accountName = accountName
        self.secret = secret
        self.algorithm = algorithm
        self.digits = digits
        self.period = period
    }
}
